crates/autopilot/src/solvable_orders.rs

```diff
@@ -604,39 +611,39 @@ async fn update_task(
     }
 }
 
-fn get_orders_with_native_prices(
+async fn get_orders_with_native_prices(
     orders: Vec<Order>,
     native_price_estimator: &CachingNativePriceEstimator,
     metrics: &Metrics,
+    additional_tokens: Vec<H160>,
+    timeout: Duration,
 ) -> (Vec<Order>, BTreeMap<H160, U256>) {
     let traded_tokens = orders
         .iter()
         .flat_map(|order| [order.data.sell_token, order.data.buy_token])
-        .collect::<HashSet<_>>()
-        .into_iter()
-        .collect::<Vec<_>>();
+        .chain(additional_tokens.into_iter())
```

Comment:
nit: Can sometimes be good to know that you don't have to call `.into_iter()` yourself when you try to make `cargo fmt` leave everything in a single line. (doesn't apply here but good to know regardless).
```suggestion
        .chain(additional_tokens)
```